engine/src/main/java/graphql/nadel/engine/execution/ServiceResultNodesToOverallResult.java

```diff
@@ -595,7 +595,19 @@ public TraversalControl enter(TraverserContext<ExecutionResultNode> context) {
     }
 
     private List<String> getFieldIdsWithoutTransformationId(ExecutionResultNode node, TransformationMetadata transformationMetadata) {
-        return node.getFieldIds().stream().filter(fieldId -> FieldMetadataUtil.getTransformationIds(fieldId, transformationMetadata.getMetadataByFieldId()).size() == 0).collect(Collectors.toList());
```

Comment:
when we have a top level field rename, child fields will have rename transformations for them. Since the field has a transformation for it, `getFieldIdsWithoutTransformationId` is empty and the `ExecutionResultNode treeWithout` in `splitTreeByTransformationDefinition` will be null.